webapp/channels/src/packages/mattermost-redux/src/reducers/entities/posts.ts

```diff
@@ -281,17 +281,14 @@ export function handlePosts(state: IDMappedObjects<Post> = {}, action: MMReduxAc
 
     case ChannelTypes.RECEIVED_CHANNEL_DELETED:
     case ChannelTypes.DELETE_CHANNEL_SUCCESS:
-    case ChannelTypes.LEAVE_CHANNEL: {
```

Comment:
This appears to have been wrong behaviour for `LEAVE_CHANNEL`: we should always discard the posts for `LEAVE_CHANNEL`. We do nothing now for `RECEIVED_CHANNEL_DELETED` and `DELETE_CHANNEL_SUCCESS`.